src/shape.rs

```diff
@@ -11,6 +11,46 @@ use unicode_segmentation::UnicodeSegmentation;
 use crate::fallback::FontFallbackIter;
 use crate::{Align, AttrsList, CacheKey, Color, Font, FontSystem, LayoutGlyph, LayoutLine, Wrap};
 
+/// The shaping strategy of some text.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum Shaping {
```

Comment:
Suggestion: I feel like this ought to be called `ShapingStrategy` or `ShapingAlgorithm` or similar rather than just `Shaping`.

Reply:
I think "basic shaping" and "advanced shaping" do make sense in a text library without any additional context.

Reply:
I agree with hecj here that there really is no need to add more context to it if its just determining from two types of shaping. However If we were to further expand this ShapingStrategy would make the most logical sense here. This would allow use to go outside the Basic or Advanced scope and allow use to implement shaping strategy's from other libraries.